gen/nvml/lib_test.go

```diff
@@ -134,6 +134,43 @@ func TestLookupFromDefault(t *testing.T) {
 	}
 }
 
+func TestLoadAndCloseNesting(t *testing.T) {
+	var numOpens, numCloses int
+
+	dl := &dynamicLibraryMock{
+		OpenFunc: func() error {
+			numOpens++
```

Comment:
I don't think this is required. You could use `len(dl.calls.Open)` and `len(dl.calls.Close)` directly.

Reply:
```
diff --git a/pkg/nvml/lib_test.go b/pkg/nvml/lib_test.go
index a800b7e..ad5a01e 100644
--- a/pkg/nvml/lib_test.go
+++ b/pkg/nvml/lib_test.go
@@ -135,15 +135,11 @@ func TestLookupFromDefault(t *testing.T) {
 }
 
 func TestLoadAndCloseNesting(t *testing.T) {
-	var numOpens, numCloses int
-
 	dl := &dynamicLibraryMock{
 		OpenFunc: func() error {
-			numOpens++
 			return nil
 		},
 		CloseFunc: func() error {
-			numCloses++
 			return nil
 		},
 	}
@@ -153,22 +149,22 @@ func TestLoadAndCloseNesting(t *testing.T) {
 	GetLibrary()
 
 	// When calling load twice, the library was only opened once
-	require.Equal(t, 0, numOpens)
+	require.Equal(t, 0, len(dl.calls.Open))
 	require.Nil(t, libnvml.load())
-	require.Equal(t, 1, numOpens)
+	require.Equal(t, 1, len(dl.calls.Open))
 	require.Nil(t, libnvml.load())
-	require.Equal(t, 1, numOpens)
+	require.Equal(t, 1, len(dl.calls.Open))
 
 	// Only after calling close twice, was the library closed
-	require.Equal(t, 0, numCloses)
+	require.Equal(t, 0, len(dl.calls.Close))
 	require.Nil(t, libnvml.close())
-	require.Equal(t, 0, numCloses)
+	require.Equal(t, 0, len(dl.calls.Close))
 	require.Nil(t, libnvml.close())
-	require.Equal(t, 1, numCloses)
+	require.Equal(t, 1, len(dl.calls.Close))
 
 	// Calling close again doesn't attempt to close the library again
 	require.Nil(t, libnvml.close())
-	require.Equal(t, 1, numCloses)
+	require.Equal(t, 1, len(dl.calls.Close))
 }
 
 func setNewDynamicLibraryDuringTest(dl dynamicLibrary) func() {
```